src/libraries/System.Text.Json/src/System/Text/Json/Serialization/Metadata/JsonPropertyInfoOfT.cs

```diff
@@ -168,7 +168,7 @@ internal sealed class JsonPropertyInfo<T> : JsonPropertyInfo
             }
             else
             {
-                Get = getter;
+                Get = getter!;
```

Comment:
`Get` is already declared as:

```C#
internal Func<object, T>? Get { get; set; }
```

Why is the `!` necessary here?

```suggestion
                Get = getter;
```

Reply:
It was necessary to assign a value of type `Func<object, T?>?` to a property of type `Func<object, T>?`.

Reply:
Shouldn't the property be changed to `Func<object, T?>?` then?

Reply:
I tried to change it, but ran into issues - https://github.com/dotnet/runtime/pull/57178#discussion_r687152012. The current implementation is a workaround.